### ISSUE TITLE
Historical data is difficult to interpret, you wont believe number 5

### ISSUE THREAD

**Issue by mvollmer** [martinpitt/performance-graphs#75]
This is general feedback after trying out historical data for a couple of minutes, please take with salt.

![image](https://user-images.githubusercontent.com/3228183/94547164-93cc9280-0257-11eb-8855-c502c131db57.png)

Above is the historical data display for compiling Cockpit in my main development VM.

I find that there are too many events. Almost every minute has an event, and it looks like just using the network in a normal bursty way counts as a spike.
The swap event is interesting and kinda unexpected, but it is easily overlooked.
I can't easily say what the two halves of the purple CPU graph represent. I think left might be user CPU, and right system CPU. Is load displayed in the graphs?
The spacing of the horizontal ticks seems random and I don't know what it represents. One minute? Why are they not evenly spaced? To accomodate the event descriptions?

**Comment by martinpitt** [comment]
Thanks for your feeback! Indeed I agree that having that many successive events is not useful. We should rate-limit them somehow -- maybe cut them off after the 5th event (of a particular resource), or maybe even better, a "dead zone" after an event? I.e. if a CPU spike happens in minute N, then no further CPU spikes are show until minute N+10, so that we get max. 6 spike events per hour. Of course *other* resource events could still happen.

Indeed we need to label the graphs properly. The left is "utilization" (how much of available resource is being used), the right is "saturation" (how much extra demand is there that cannot be satisfied right away). For CPU, that's usage (left) vs load (right), for RAM that's again usage (left) vs. swap (right), and disk/network don't currently have saturation (see issue #60).

The horizontal tick marks are one minute, yes. Eventful minutes are stretched so that you can see the graphs better at these, and the "uneventful" ones are compressed. I was hoping that the times in the events would give a sufficient labeling, and the tooltips also contain the precise time including seconds.

**Comment by martinpitt** [comment]
Adding @andreasn and @garrett, as this is a design discussion again.

**Comment by andreasn** [comment]
Agree about some kind of labels being needed for saturation and utilization. It's a novel design, so people aren't used at looking at these, hence having no or little prior knowledge to fall back on.

Very crude mockup, but maybe something like this?
![image](https://user-images.githubusercontent.com/209895/94658213-ced7d000-0302-11eb-9add-11c2ea7c2300.png)

**Comment by marusak** [comment]
> Very crude mockup, but maybe something like this?

The labels are the other way around (just saying) but I am more concern that "saturation" does not say much to an average user. I like the explanation that Martin provided. Can we get it somewhere as well? (like (?) icon with tooltip or at least in docs and then link to that somehow?)

**Comment by martinpitt** [comment]
Yes, I didn't intend to literally say utilization/saturation -- that's just a conceptual description. Let's put the actual metric names there. "CPU Usage | Load" for the first column, and "RAM Usage | Swap out" for the second? We can keep "Disks" and "Network" for third and fourth for now (see issue #60)

**Comment by andreasn** [comment]
![image](https://user-images.githubusercontent.com/209895/94659763-12333e00-0305-11eb-8925-8dbe17f76480.png)

**Comment by mvollmer** [comment]
>  Indeed I agree that having that many successive events is not useful.

Also, at least the "spike" events don't seem to add much value at all. I can already see how the CPU usage looks and whether there is a spike in it. Events would be super useful if the show things that are not in the graphs, like critical journal messages, reboots, etc.

**Comment by garrett** [comment]
> Events would be super useful if they show things that are not in the graphs, like critical journal messages, reboots, etc.

This was the original goal.

(I chose spikes as filler text, basically.)

**Comment by mvollmer** [comment]
> (I chose spikes as filler text, basically.)

Ok, let's just remove spike detection altogether then, non?

**Comment by garrett** [comment]
I think spike detection is still useful. We'd probably want that area of the graph more detailed still? Spiking is probably more useful for the saturation rather than utilization, however. That is, CPU load spiking, not the usage, and hitting swap instead of just using RAM or slowly using swap.

**Comment by marusak** [comment]
Labels has been updated, so this issue is now about "Too many spikes" and "Spikes should provide more info"

**Comment by garrett** [comment]
The information should be about other events, not necessarily about the spikes.

How / why are spikes useful?
- In most cases, a momentary blip is usually less important than a sustained hit over (a few seconds or more), I think?
- You want to know what process is hitting CPU, gobbling memory, hammering the storage, and flooding the network — if one process is doing more than one of these things at a time, then it's all the more important to show that
- It might be good to know how a resource "hungry" process compares to its typical state (its mean, median, or other normal resource usage)
